account/local_account.go

```diff
@@ -13,41 +14,101 @@ import (
 
 // LocalAccount represents a local account implementation
 type LocalAccount struct {
-	*key.SoftKey
+	name    string
+	softKey *key.SoftKey
 }
 
-// NewLocalAccount creates a new LocalAccount
-func NewLocalAccount() (Account, error) {
+// NewLocalAccount creates a new local account with a freshly generated private key
+func NewLocalAccount(name string) (*LocalAccount, error) {
 	k, err := key.NewSoft()
 	if err != nil {
 		return nil, err
 	}
 	return &LocalAccount{
-		SoftKey: k,
+		name:    name,
+		softKey: k,
 	}, nil
 }
 
-func Import(keyPath string) (Account, error) {
-	k, err := key.LoadSoft(keyPath)
+// NewLocalAccountFromPrivateKey imports a local account from a hex-encoded private key string
+func NewLocalAccountFromPrivateKey(name string, privateKeyHex string) (*LocalAccount, error) {
+	k, err := key.LoadSoftFromBytes([]byte(privateKeyHex))
 	if err != nil {
 		return nil, err
 	}
 	return &LocalAccount{
-		SoftKey: k,
+		name:    name,
+		softKey: k,
 	}, nil
 }
 
+// NewLocalAccountFromKeyPath imports a local account from a file path containing a private key
+func NewLocalAccountFromKeyPath(name string, keyPath string) (*LocalAccount, error) {
+	kb, err := os.ReadFile(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	return NewLocalAccountFromPrivateKey(name, string(kb))
+}
+
+func (a *LocalAccount) Name() string {
+	return a.name
+}
+
 func (a *LocalAccount) GetPChainAddress(network network.Network) (string, error) {
-	if a.SoftKey == nil {
-		return "", fmt.Errorf("SoftKey not initialized")
+	if a.softKey == nil {
```

Comment:
I think we should check if local account is not nil here instead of checking the softKey, since all fields are private in the struct and we expose the init functions to create the struct with all fields set.

Reply:
same in other methods below